### ISSUE TITLE
Lighthouse cannot get a Performance score ~30% of the time

### ISSUE THREAD

**Issue by wildlyinaccurate** [GoogleChrome/lighthouse#9107]
Hey folks! At SpeedCurve we're running Lighthouse about 100,000 times every day across almost as many distinct URLs. We consider a LH run to have failed if it doesn't return a Performance score. We have a baseline failure rate of 8-9% (mostly due to bot managers) but we noticed that around April 25 the failure rate jumped to about 30%.

![Percent Failed vs Date](https://user-images.githubusercontent.com/794263/58775124-26cfcb00-8618-11e9-93c7-1a900938902c.png)

> The raw data for this chart is [in a public spreadsheet](https://docs.google.com/spreadsheets/d/1p8KPtbm_mWhQi96N8J3ra5qm9NoaatzyBJiafFqfOHc/edit?usp=sharing) if you want to dive into it more. You'll notice the failure rate is less for mobile than it is for desktop.

We don't track the failure reason, but anecdotally the NO_SCREENSHOTS error is pretty common. [Here's an example test for a URL that gets this error roughly 1 in 3 runs](https://wpt.speedcurve.com/lighthouse.php?test=190601_N8_3f60d565ad49065e20b23e7f4f6326ba&run=2). I didn't see this error often before 25 April.

#### Environment Information

SpeedCurve runs its Lighthouse testing via WebPageTest. WPT performs its own throttling, using tc-netem for the network and Chrome DevTools for the CPU.

* Affected Channels: CLI
* Lighthouse version: 5.1.0
* Node.js version: 10.x
* Operating System: Ubuntu 18.04

Example command that would be run by wptagent:

```sh
# {PORT} is the port of an already-running Chrome instance, potentially with CPU throttling applied.
lighthouse "{URL}" --disable-network-throttling --disable-cpu-throttling --throttling-method provided --enable-error-reporting --max-wait-for-load 60000 --port {PORT} --output html --output json --output-path "{PATH}"
```

**Comment by brendankenny** [comment]
@wildlyinaccurate huge thanks for this data.

We've also been encountering a lot more `NO_SCREENSHOTS` but it's been hard to pin down exactly how frequently it's happening for real runs (we've seen it sometimes in the CLI, for instance, but not that often. And the extension tests in CI fail often on it while all the CLI tests in the same run in the same browser will happily pass, etc, etc). The failure rate you're seeing is much higher than we've experienced, though, and definitely unexpected (cc @exterkamp).

Right now our working theory is that it was a regression in Chrome tracing, but we haven't figured out what happened yet. @paulirish bisected a problem to https://crbug.com/966681, but given that that appears to be Mac only, I don't think we have our culprit.

Chrome 74 was released to stable on April 23rd (with a staged rollout over several days). Do you usually test against stable?

(see also #9027)

**Comment by exterkamp** [comment]
To add some color on the Lighthouse-at-scale problem: This error rate seems high.  We do on occasion see spikes like this that can last for _maybe_ a day from unexplained means, but for it to be permanent is unexpected.

I would say an error rate of 10-15% is not unexpected due to the background radiation of people entering in URLs that are not real, pages not loading intermittently, etc.  

I'm curious if you get the same kinds of results with the [PSI API](https://developers.google.com/speed/pagespeed/insights/).  It's probably too quick to say that it's all localized to the WPT runtime, but maybe their combination of LH + Chrome version is not ideal.  On my end I definitely don't see `NO_SCREENSHOTS` as a major error from our stack.

**Comment by Hoten** [comment]
WPT uses stable. You can verify by looking at LH reports they generate, and seeing the UA at the bottom.

**Comment by zeman** [comment]
@brendankenny For more context the WebPageTest agents reboot every hour and update the browsers and Lighthouse to latest release/stable version and that's what we test with.

**Comment by wildlyinaccurate** [comment]
@brendankenny thanks for the info. My next step will be to find individual URLs that have exceptionally high failure rates, and see if they can be used to reliably reproduce some of the issues. Are there any artefacts I could collect for these URLs that would be useful for you?

@exterkamp we stopped using the PSI API after deprecating PageSpeed in favour of Lighthouse, so I don't have any recent data to compare against. Do you know if the PSI API does any result caching or does it make a fresh LH run every time?

**Comment by brendankenny** [comment]
I totally breezed over the fact that you're using WPT for the test runs; sorry about that :)

> My next step will be to find individual URLs that have exceptionally high failure rates, and see if they can be used to reliably reproduce some of the issues. Are there any artefacts I could collect for these URLs that would be useful for you?

The screenshots are extracted from the trace directly, so just the trace is sufficient.

If you need a quick local test (without running all of Lighthouse), something like `trace.traceEvents.some(e => e.name === 'Screenshot')` should be able to detect if a trace is good or will throw the `NO_SCREENSHOTS` error

**Comment by exterkamp** [comment]
😭 that you moved off PSI.  It does do some results caching within the same minute iirc.  But ordinarily it makes a fresh LH run each time. @paulirish might have more color on that & our intentions around caching.

PSI's evergreen canary env also doesn't see a huge error rate like you're seeing.  Which seems odd based on #9027 saying it should happen everytime with m74+ & LR 4.3+?

**Comment by wildlyinaccurate** [comment]
@exterkamp we moved away from PSI on the understanding that PSI v5 is essentially a wrapper around Lighthouse, at least as far as the overall score is concerned. It didn't make sense for us to be making external calls when we were able to get the same data internally. Did we get the wrong end of the stick on that?

Are you able to share some information about PSI's evergreen canary environment? OS info, CLI or extension, etc? It might help to highlight any differences with how we run LH (other than via WPT). #9027 is for macOS; we only run Lighthouse from our Linux test agents.
